### PR TITLE
ui-spacetimechart: add conflict tooltips

### LINKED ISSUE #418: spacetimechart: add conflicts tooltips

### PR DESCRIPTION
**Pull request by emersion**
_See individual commits._

Mockup: https://www.sketch.com/s/bcbac4ff-7f74-4f1a-9e10-08d115c47ed8/p/93690D83-1F83-4AC1-B403-560752B06658/canvas#Inspect

Closes: https://github.com/OpenRailAssociation/osrd-ui/issues/418

### CODE REVIEW

**Comment by emersion** [comment]
This is feature complete, but leaving as a draft PR since I have one UX question: I don't think it makes sense to have multiple conflicts listed in the tooltip. Will ask around. Shouldn't have a lot of impact either way.

File: ui-spacetimechart/src/lib/types.ts [hunk 1]
```diff
@@ -99,7 +99,8 @@ export type LayerType = (typeof LAYERS)[number];
 // PICKING SPECIFIC TYPES:
 export type PickingElement =
   | { type: 'point'; pathId: string; point: Point }
-  | { type: 'segment'; pathId: string; from: Point; to: Point };
+  | { type: 'segment'; pathId: string; from: Point; to: Point }
+  | { type: 'conflict'; conflictIndex: number };
```
Comment (emersion) [inline ui-spacetimechart/src/lib/types.ts:103]: @jacomyal Was this how you envisioned how picking would work for other kinds of elements? Or should we try to make picking more modular?

**Review by Synar — APPROVED** [review]
Nice, LGTM

I will try do another pass tomorrow to find something to nitpick about though ^^

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark:

**Comment by emersion** [comment]
Updated to follow the latest mockup. Also clarified that the time displayed in green text is wired up to the cursor, it's not the conflict start time. (Reviewers: only the last commit has been edited.)

File: ui-spacetimechart/src/styles/main.css [hunk 2]
```diff
@@ -1,3 +1,46 @@
...
+    line-height: 20px;
+    text-align: center;
+    margin-bottom: 9px;
+  }
+
+  .position-range {
+    display: flex;
+    justify-content: space-between;
+    font-size: 0.75rem;
+    margin-bottom: 13px;
```
Comment (clarani) [inline ui-spacetimechart/src/styles/main.css]:
```suggestion
    margin-bottom: 15px;
```

File: ui-spacetimechart/src/styles/main.css [hunk 3]
```diff
@@ -1,3 +1,46 @@
...
+
+  .type-and-duration, .trains {
+    font-weight: 600;
+  }
+
+  .type-and-duration {
+    display: flex;
+    justify-content: space-between;
+    background-color: theme('colors.error.60');
+    padding: 3px 6px;
```
Comment (clarani) [inline ui-spacetimechart/src/styles/main.css]:
```suggestion
    padding: 1px 6px 3px 6px;
```

Comment (clarani) [inline ui-spacetimechart/src/styles/main.css]: Margin bottom is missing:
```
    margin-bottom: 2px;
```

File: ui-spacetimechart/src/styles/main.css [hunk 4]
```diff
@@ -1,3 +1,46 @@
...
+  padding: 6px 12px 7px 12px;
+  margin: 12px;
+  width: 213px;
+
+  .time {
+    text-shadow: 0 0 6px rgba(27, 255, 122, 1);
+    color: rgba(27, 255, 122, 1);
+    line-height: 20px;
+    text-align: center;
+    margin-bottom: 9px;
```
Comment (clarani) [inline ui-spacetimechart/src/styles/main.css]:
```suggestion
    margin-bottom: 3px;
```

File: ui-spacetimechart/src/styles/main.css [hunk 5]
```diff
@@ -1,3 +1,46 @@
...
+
+  .position-range {
+    display: flex;
+    justify-content: space-between;
+    font-size: 0.75rem;
+    margin-bottom: 13px;
+  }
+
+  .type-and-duration, .trains {
+    font-weight: 600;
```
Comment (clarani) [inline ui-spacetimechart/src/styles/main.css]: Line height is not correct
```
    line-height: 20px;
```

File: ui-spacetimechart/src/styles/main.css [hunk 6]
```diff
@@ -1,3 +1,46 @@
...
+
+  .time {
+    text-shadow: 0 0 6px rgba(27, 255, 122, 1);
+    color: rgba(27, 255, 122, 1);
+    line-height: 20px;
+    text-align: center;
+    margin-bottom: 9px;
+  }
+
+  .position-range {
```
Comment (clarani) [inline ui-spacetimechart/src/styles/main.css:28]: Line-height is not correct:
```
    line-height: 16px;
```

**Review by clarani — COMMENTED** [review]
- The positions should be bold
- Do you think you can replace the `,` by `.` for the positions to follow the mockup ?
image

Mockup:
image

**Comment by emersion** [comment]
Thanks for the detailed comments, many things I've missed! I'm not super sure all of these details were really intentional UX decisions, but oh well.

> Do you think you can replace the , by . for the positions to follow the mockup ?

Ah yes, I've been formatting numbers according to the current locale, but it seems the expectation is to always use English-style floating point number formatting here… Fixed by forcing the locale to US.

File: ui-spacetimechart/src/styles/main.css [hunk 7]
```diff
@@ -1,3 +1,48 @@
...
+.spacetimechart-tooltip {
+  position: absolute;
+  background-color: rgba(0, 0, 0, 0.85);
+  box-shadow: 0 4px 6px -2px rgba(0, 0, 0, 0.28);
+  color: white;
+  font-size: 0.875rem;
+  line-height: 20px;
+  border-radius: 10px;
+  padding: 6px 12px 7px 12px;
+  margin: 12px;
```
Comment (clarani) [inline ui-spacetimechart/src/styles/main.css:14]: Why did you add this margin ?

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark:

File: ui-spacetimechart/src/styles/main.css (hunk 7 above)
Comment (emersion) [reply ui-spacetimechart/src/styles/main.css:14]: It's used to offset the tooltip from the cursor. Without it, the tooltip's top-left corner would appear directly under the cursor.
